routing/endpointregistry.go

```diff
@@ -14,10 +14,13 @@ const lastSeenTimeout = 1 * time.Minute
 type Metrics interface {
 	DetectedTime() time.Time
 	InflightRequests() int64
+	FadeInDuration() time.Duration
 }
 
 type entry struct {
 	detected         time.Time
+	fadeInDuration   time.Duration
```

Comment:
In general, one endpoint could be used in several routes. So, it makes sense to create 1-to-many relation here somehow. The only thing that comes to my mind at the moment is map from routeId to struct with duration and exponent.

Reply:
Start simple, add metrics, measure, observe, optimize.